### PR TITLE
Fix memory leak in talkaction

### PR DESCRIPTION
**Pull request by Mkalo**
That was my mistake when changing it to list of objs instead of pointers.

### CODE REVIEW

File: src/talkaction.cpp [hunk 1]
```diff
@@ -61,7 +61,9 @@ Event* TalkActions::getEvent(const std::string& nodeName)
 
 bool TalkActions::registerEvent(Event* event, const pugi::xml_node&)
 {
-	talkActions.push_front(*static_cast<TalkAction*>(event)); // event is guaranteed to be a TalkAction
+	TalkAction* talkAction = static_cast<TalkAction*>(event); // event is guaranteed to be a TalkAction
+	talkActions.push_front(*talkAction);
```
Comment (ranisalt) [inline src/talkaction.cpp]: `std::move(*talkaction)` might use less resources than copying here. Also I'd use an `unique_ptr` for clean pointer handling.

```
auto talkaction = std::unique_ptr<Talkaction>(*event);
talkactions.push_front(std::move(*talkaction.get()));
```

File: src/talkaction.cpp [hunk 2]
```diff
@@ -61,7 +61,8 @@ Event* TalkActions::getEvent(const std::string& nodeName)
 
 bool TalkActions::registerEvent(Event* event, const pugi::xml_node&)
 {
-	talkActions.push_front(*static_cast<TalkAction*>(event)); // event is guaranteed to be a TalkAction
+	auto talkAction = std::unique_ptr<TalkAction>(static_cast<TalkAction*>(event)); // event is guaranteed to be a TalkAction
+	talkActions.push_front(std::move(*talkAction.get()));
```
Comment (ranisalt) [inline src/talkaction.cpp]: Just noticed you can use `*talkAction`, `.get()` is redundant [(source)](http://en.cppreference.com/w/cpp/memory/unique_ptr/operator*)

Comment (Kamenuvol) [inline src/talkaction.cpp]: I'd rather use get to emphasize is not a normal pointer

**Comment by ranisalt** [comment]
Thanks!

**Pull request merged**